examples/samples/deferral.ts

```diff
@@ -0,0 +1,130 @@
+} from "../../lib";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+const str = process.env.SERVICEBUS_CONNECTION_STRING || "";
+const path = process.env.QUEUE_NAME || "";
+console.log("str: ", str);
+console.log("path: ", path);
+
+async function main(): Promise<void> {
+  await sendMessage();
+  await receiveMessage();
+}
+
+async function sendMessage(): Promise<void> {
+  const nsSend = Namespace.createFromConnectionString(str);
+  const sendClient = nsSend.createQueueClient(path);
+  var data = [
+    { step: 1, title: "Shop" },
+    { step: 2, title: "Unpack" },
+    { step: 3, title: "Prepare" },
+    { step: 4, title: "Cook" },
+    { step: 5, title: "Eat" }
+  ];
+  var promises = new Array();
+  for (let index = 0; index < data.length; index++) {
+    const message: SendableMessageInfo = {
+      body: data[index],
+      label: "RecipeStep",
+      contentType: "application/json",
+      timeToLive: 2 * 60 * 1000, // 2 minutes
+      messageId: generateUuid()
+    };
+    // the way we shuffle the message order is to introduce a tiny random delay before each of the messages is sent
+    promises.push(
+      delay(Math.random() * 30).then(async () => {
+        try {
+          await sendClient.send(message);
+          console.log("Sent message step:", data[index].step);
+        } catch (err) {
+          console.log("Error while sending message", err);
+        }
+      })
+    );
+  }
+  // wait until all the send tasks are complete
+  await Promise.all(promises);
+  nsSend.close();
+}
+
+async function receiveMessage(): Promise<void> {
+  const nsRcv = Namespace.createFromConnectionString(str);
+  const receiveClient = nsRcv.createQueueClient(path, { receiveMode: ReceiveMode.peekLock });
+  var deferredSteps = new Map();
+  var lastProcessedRecipeStep = 0;
+  try {
+    const onMessage: OnMessage = async (brokeredMessage: ServiceBusMessage) => {
+      if (
+        brokeredMessage.label === "RecipeStep" &&
+        brokeredMessage.contentType === "application/json"
+      ) {
+        const message = brokeredMessage.body;
+        // now let's check whether the step we received is the step we expect at this stage of the workflow
+        if (message.step == lastProcessedRecipeStep + 1) {
+          console.log("Message Received:", brokeredMessage.body ? message : null);
+          lastProcessedRecipeStep++;
+          await brokeredMessage.complete();
+        } else {
+          // if this is not the step we expected, we defer the message, meaning that we leave it in the queue but take it out of
+          // the delivery order. We put it aside. To retrieve it later, we remeber its sequence number
+          const sequenceNumber = brokeredMessage.sequenceNumber;
+          deferredSteps.set(message.step, sequenceNumber);
+          await brokeredMessage.defer();
+        }
+      } else {
+        // we dead-letter the message if we don't know what to do with it.
+        brokeredMessage.deadLetter();
```

Comment:
We should add an `await` here. And move the console.log above this line

Reply:
Ok.